### PR TITLE
change string concatenation to string formatting in inventory/digital…

### PR DESCRIPTION
**Pull request by rusik69**
##### SUMMARY

change string concatenation to string formatting in inventory/digital_ocean.py

##### ISSUE TYPE

 - Docs Pull Request

##### COMPONENT NAME

contrib/inventory/digital_ocean.py

##### ANSIBLE VERSION

```
2.4.0
```

##### ADDITIONAL INFORMATION

```

```

### CODE REVIEW

File: contrib/inventory/digital_ocean.py [hunk 1]
```diff
@@ -193,7 +193,7 @@ def __init__(self):
             sys.exit(0)
 
         # Manage cache
-        self.cache_filename = self.cache_path + "/ansible-digital_ocean.cache"
```
Comment (Akasurde) [inline contrib/inventory/digital_ocean.py:196]: Any specific reason for this change ?

File: contrib/inventory/digital_ocean.py [hunk 2]
```diff
@@ -250,7 +250,7 @@ def __init__(self):
     def read_settings(self):
         ''' Reads the settings from the digital_ocean.ini file '''
         config = ConfigParser.SafeConfigParser()
-        config.read(os.path.dirname(os.path.realpath(__file__)) + '/digital_ocean.ini')
+        config.read('%s/digital_ocean.ini' % os.path.dirname(os.path.realpath(__file__)))
```
Comment (Akasurde) [inline contrib/inventory/digital_ocean.py:253]: Any specific reason for this change? I would prefer `os.path.join` instead.

File: contrib/inventory/digital_ocean.py [hunk 3]
```diff
@@ -273,10 +273,10 @@ def read_settings(self):
     def read_environment(self):
         ''' Reads the settings from environment variables '''
         # Setup credentials
-        if os.getenv("DO_API_TOKEN"):
-            self.api_token = os.getenv("DO_API_TOKEN")
-        if os.getenv("DO_API_KEY"):
-            self.api_token = os.getenv("DO_API_KEY")
+        if os.getenv('DO_API_TOKEN'):
```
Comment (Akasurde) [inline contrib/inventory/digital_ocean.py:276]: why change `"` to `'` ?

**Review by Akasurde — CHANGES_REQUESTED** [review]

File: contrib/inventory/digital_ocean.py (hunk 2 above)
Comment (alikins) [reply contrib/inventory/digital_ocean.py:253]: I agree with a preference for os.path.join() if we are changing it.

**Comment by alikins** [comment]
> change string concatenation to string formatting in inventory/digital_ocean.py

Was the string concat causing problems? I generally prefer the string formatting (or os.path.join() for paths) over concat but curious if the current code was breaking something and commit message or pr text doesn't mention why.

**Comment by bcoca** [comment]
FYI, The project preference is concatenation as formatting has security issues

**Pull request closed without merging**